SonarQube.MSBuild.Tasks/WriteProjectInfoFile.cs

```diff
@@ -79,8 +82,27 @@ public override bool Execute()
             pi.FullPath = this.FullProjectPath;
             pi.ProjectLanguage = this.ProjectLanguage;
 
+            string guid = null;
+            if (!String.IsNullOrEmpty(this.ProjectGuid))
+            {
+                guid = this.ProjectGuid;
+            }
+            else if (!String.IsNullOrEmpty(this.SolutionConfigurationContents))
+            {
+                // Try to get GUID from the Solution
+                XmlDocument xmlDoc = new XmlDocument();
```

Comment:
It mainly depends if you can have multiple times the same attribute for a given node.